lib/services/access_request.go

```diff
@@ -2408,7 +2423,15 @@ func (m *RequestValidator) pruneResourceRequestRoles(
 				`no roles configured in the "search_as_roles" for this user allow `+
 					`access to at least one requested resources. `+
 					`resources: %s roles: %v unmatched resources: %v`,
-				resourcesStr, roles, kubeResourceMatcher.Unmatched())
+				resourcesStr, requestedRoles, kubeResourceMatcher.Unmatched())
+		}
+		if resource.GetKind() == types.KindSAMLIdPServiceProvider {
+			// Access precheck to validate if there is a restricting standing
+			// role for this user, which may otherwise deny access to this
+			// resource even after assuming roles granted by this access request.
+			if err := canAccessSaml(localClusterName, roleSet, resource, m.userState.GetTraits(), matchers...); err != nil {
+				return nil, trace.Wrap(err)
+			}
```

Comment:
This seems like it would return an error if it is merely possible to request a role that would not allow access to the requested SAMLIdPServiceProvider, where the desired behaviour would be for that role to just be filtered out and not requested